packages/flutter/lib/src/painting/borders.dart

```diff
@@ -151,20 +162,32 @@ class BorderSide {
   /// A hairline black border that is not rendered.
   static const BorderSide none = BorderSide(width: 0.0, style: BorderStyle.none);
 
-  /// The direction of where the border will be drawn relative to the container.
-  final StrokeAlign strokeAlign;
+  /// The relative position of the stroke on a [BorderSide] in an
+  /// [OutlinedBorder] or [Border].
+  ///
+  /// Values typically range from -1.0 (inside border, default) to
+  /// 1.0 (outside border), without any bound constraints (e.g., a
+  /// value of -2.0 is is not typical, but allowed).
+  ///
+  /// See also:
+  ///
+  ///  * [StrokeAlign], a class that acts as a shortcut for specifying
```

Comment:
I realize that this is a source of controversy, but I think the need to document the values of StrokeAlign's constants here supports the idea that we'd be better off with comparable constants defined on BorderSide.  Defining a class just to enshrine 3 constants that could be given equally evocative names and sufficient API doc here, doesn't seem like a good trade. 

That said, this a great PR; adds a nice dimension of flexibility to BorderSide.

Reply:
Hey, Hans! Thanks for the review! I'm feeling the pressure for BorderSide to return. lol

I'll explain my thinking, if you disagree, I'll revert to `BorderSide.strokeAlignInside`, ok?
My rationale was:

Pros for StrokeAlign.inside:
- Do not break existing API (either way wouldn't be breaking, StrokeAlign is barely in Flutter beta and no one uses yet).
- Development experience is a lot easier. There is no autocomplete confusion, the word to type is much shorter, ctrl+shift+back + 'center' just works, co-pilot gets less on the way. If you type `c` it suggests `center` instead of `canLerp`.
- Discoverability is better. There is `style: BorderStyle.solid` and `color: Color()`, so `strokeAlign: StrokeAlign.inside` is natural.

Hypothetically, thinking about the future:
- If eventually, we go from `double` to `Offset` (I have the PR mostly ready for two-dimension StrokeAlign, but it lacks a reason to exist, so don't expect it soon), the migration path is enum -> double -> offset, with the same `StrokeAlign.inside` working always.  `BorderSide.strokeAlignInside` takes us to enum -> BorderSide -> (deprecation?) -> Offset.
- If we get the multi-directional BorderSide (PR almost ready, need to convince the team first it is a good idea), there are two possible implementations: `BorderSide.only()` (I prefer) and `MultiBorderSide extends BorderSide`. If we go with the second, `MultiBorderSide.strokeAlignInside` would be possible, which is also odd.

These two are pure speculation, I am just trying to look at what the future holds depending on the decision we take here.

Pros for `BorderSide.strokeAlignCenter`:
- You avoid allocating a class that doesn't need to exist.
- Theoretically should be slightlyyyy 0.0001s faster due to this.
- Behaves like other Flutter APIs (which is why I'm trying to go 2D and justify StrokeAlign lol).

I think everyone agrees `BorderSide.strokeAlignCenter` is ugly but more efficient. I think the same. We've gone forth and back a few times in this PR. I think a single class allocation could be forgiven due to the benefits in development performance. Is it 0.0001s slower? Possibly. But I think the development advtange pays the cost.

That said, do you still prefer `BorderSide.strokeAlignInside`?

Reply:
2x1, but speed is important, so it is a tie. I'll let you decide 🤣.

<img width="523" alt="image" src="https://user-images.githubusercontent.com/351125/182285641-dfa50def-981c-471e-bb73-7bbb348d2e36.png">

Reply:
Thank you for the comprehensive analysis and for the nicely illustrated advantages in terms for Flutter's core values. I don't think I've seen a perspective on a technical decision expressed quite like this before.

I think a 2D strokeAlign parameter would enable some interesting effects, and it's a creative way to extend the stroke alignment idea, but I don't think it's the ideal way to generalize the definition of stroke width. That seems like it needs to be a function of distance along the border, like a Bezier curve's _t_ parameter. I don't think that kind of generalization is a feature that would benefit most developers. At least not now.

I do still prefer BorderSide.strokeAlignInside et al. I appreciate your passion about this feature and I hope we can make this small adjustment and get this PR landed. It's a nice piece of work.

Reply:
>  That seems like it needs to be a function of distance along the border, like a Bezier curve's t parameter.

That sounds fascinating, could you explain more? Maybe I can try later. Right now, the 2D StrokeAlign is a different StrokeAlign for each side. So you can have left side set to -1, right side to center (0), and right side to outside (1). It works well, symmetric shapes can make a nice effect. Maybe I can start a document proposal and we can think together. Another day. Right now I'm writing [this proposal](https://docs.google.com/document/d/1rEQhlPzUj1QPEGSiA7eLl6kEqlkJFUecpKn-9Hcsddc/edit?usp=sharing&resourcekey=0-fpMavpYbDkqrkW2iNclV_Q). Hixie didn't like `BorderSide.only`, so I'm thinking about rewriting Border (and if it is well accepted, then fix BorderSide or maybe never fix it). Feel free to help if you like the idea. Greg has had enough of me already.

> I do still prefer BorderSide.strokeAlignInside
Ok :(
You won. Reverted.